cmd/oras/internal/option/format.go

```diff
@@ -64,26 +64,41 @@ var (
 		Name:  "tree",
 		Usage: "Get referrers recursively and print in tree format",
 	}
+	FormatTypeText = &FormatType{
+		Name:  "text",
+		Usage: "Print in text format",
+	}
 )
 
 // Format contains input and parsed options for formatted output flags.
 type Format struct {
 	FormatFlag   string
 	Type         string
 	Template     string
-	AllowedTypes []*FormatType
+	allowedTypes []*FormatType
+	defaultType  *FormatType
+}
+
+// SetTypes sets the default format type and allowed format types.
+func (f *Format) SetTypes(defaultType *FormatType, allowedTypes ...*FormatType) {
+	f.defaultType = defaultType
+	f.allowedTypes = append(allowedTypes, defaultType)
 }
 
-// ApplyFlag implements FlagProvider.ApplyFlag.
+// ApplyFlags implements FlagProvider.ApplyFlag.
 func (opts *Format) ApplyFlags(fs *pflag.FlagSet) {
 	buf := bytes.NewBufferString("[Experimental] Format output using a custom template:")
 	w := tabwriter.NewWriter(buf, 0, 0, 2, ' ', 0)
-	for _, t := range opts.AllowedTypes {
+	for _, t := range opts.allowedTypes {
 		_, _ = fmt.Fprintf(w, "\n'%s':\t%s", t.Name, t.Usage)
 	}
-	w.Flush()
+	_ = w.Flush()
+	defaultTypeName := FormatTypeText.Name
+	if opts.defaultType != nil {
+		defaultTypeName = opts.defaultType.Name
+	}
```

Comment:
Just curious when will `defaultType` be `nil`?

Reply:
Only if someone didn't call SetTypes is all I can guess.

Reply:
Terry is right. Double checked the implementation and it seems that we can remove the type and store default value into `FormatFlag`. Refactored and repushed.